Source/WebCore/platform/gstreamer/GStreamerQuirkBroadcomBase.cpp

```diff
@@ -0,0 +1,194 @@
+{
+    auto state = ensureState(playerPrivate);
+
+    // The Nexus playpump buffers a lot of data. Let's add it as if it had been buffered by the GstQueue2
+    // (only when using in-memory buffering), so we get more realistic percentages.
+    if (mode != GST_BUFFERING_STREAM || !(state.m_vidfilter || state.m_audfilter))
+        return originalBufferingPercentage;
+
+    // The Nexus playpump buffers a lot of data. Let's add it as if it had been buffered by the GstQueue2
+    // (only when using in-memory buffering), so we get more realistic percentages.
+    int correctedBufferingPercentage1 = originalBufferingPercentage;
+    int correctedBufferingPercentage2 = originalBufferingPercentage;
+    unsigned maxSizeBytes = 0;
+
+    // We don't trust the buffering percentage when it's 0, better rely on current-level-bytes and compute a new buffer level accordingly.
+    g_object_get(state.m_queue2.get(), "max-size-bytes", &maxSizeBytes, nullptr);
+    if (!originalBufferingPercentage && state.m_queue2) {
+        unsigned currentLevelBytes = 0;
+        g_object_get(state.m_queue2.get(), "current-level-bytes", &currentLevelBytes, nullptr);
+        correctedBufferingPercentage1 = currentLevelBytes > maxSizeBytes ? 100 : static_cast<int>(currentLevelBytes * 100 / maxSizeBytes);
+    }
+
+    unsigned playpumpBufferedBytes = 0;
+
+    // We believe that the playpump stats are common for audio and video, so only asking the vidfilter or the audfilter
+    // would be enough. Both of them expose the buffered_bytes property (yes, with an underscore!).
+    GstElement* filter = state.m_vidfilter ? state.m_vidfilter.get() : state.m_audfilter.get();
+    if (filter)
+        g_object_get(GST_OBJECT(filter), "buffered_bytes", &playpumpBufferedBytes, nullptr);
+
+    unsigned multiqueueBufferedBytes = 0;
+    if (state.m_multiqueue) {
+        GUniqueOutPtr<GstStructure> stats;
+        g_object_get(state.m_multiqueue.get(), "stats", &stats.outPtr(), nullptr);
+        for (const auto& queue : gstStructureGetArray<const GstStructure*>(stats.get(), "queues"_s))
+            multiqueueBufferedBytes += gstStructureGet<unsigned>(queue, "bytes"_s).value_or(0);
+    }
+
+    // Current-level-bytes seems to be inacurate, so we compute its value from the buffering percentage.
+    size_t currentLevelBytes = static_cast<size_t>(maxSizeBytes) * static_cast<size_t>(originalBufferingPercentage) / static_cast<size_t>(100)
+        + static_cast<size_t>(playpumpBufferedBytes) + static_cast<size_t>(multiqueueBufferedBytes);
+    correctedBufferingPercentage2 = currentLevelBytes > maxSizeBytes ? 100 : static_cast<int>(currentLevelBytes * 100 / maxSizeBytes);
+
+    if (correctedBufferingPercentage2 >= 100)
+        state.m_streamBufferingLevelMovingAverage.reset(100);
+    int averagedBufferingPercentage = state.m_streamBufferingLevelMovingAverage.accumulate(correctedBufferingPercentage2);
+
+#ifndef GST_DISABLE_GST_DEBUG
```

Comment:
The ifdef is not preventing a hard and avoidable. computation, I think you could remove it.